test/client.jl

```diff
@@ -242,4 +242,22 @@ end
         test(HTTP.delete(uri, headers; body=body, query=query), "DELETE")
         test(HTTP.delete(uri, headers, body; query=query), "DELETE")
     end
+    # Setting of Content-Type with body::Form
+    body = HTTP.Form(Dict())
+    uri = "https://httpbin.org/post"
+    for r in (HTTP.request("POST", uri, headers, body), HTTP.post(uri, headers, body))
+        @test r.status == 200
+        json = JSON.parse(IOBuffer(HTTP.payload(r)))
+        @test startswith(json["headers"]["Content-Type"], "multipart/form-data; boundary=")
+    end
+    # Deprecation of HTTP.post without header for body::Form
+    proj = normpath(joinpath(pathof(HTTP), "..", "..", "Project.toml"))
+    vers = VersionNumber(match(r"^version\s*=\s*\"(.*?)\"$"m, read(proj, String)).captures[1])
```

Comment:
Can there be a comment explaining the `regex`? Makes it easier to know what you're matching at a quick glance.